### ISSUE TITLE
Sitemap Plugin

### ISSUE THREAD

**Issue by mikhin** [vikejs/vike#1451]
### Description

Hello, Vike Community! I'm in search of a sitemap plugin to enhance our project and was wondering if such a feature already exists.

If not, I'd be grateful for any guidance or suggestions on how to elegantly implement this.

Thank you in advance for your time and assistance!

**Comment by brillout** [comment]
What exactly do you mean with sitemap? Do you mean `example.org/sitemap.xml`? What's your motivation?

**Comment by mikhin** [comment]
Yeah, `sitemap.xml`. I have `ssr({prerender: true})` and a lot of html files after build phase, so I want to get somehow `sitemap.xml` after build step too.

**Comment by brillout** [comment]
How about a `prerender: { sitemap: true }` option? Would you be up for a PR?

> What's your motivation?

For SEO or other purposes as well? Is it still relevant to have a `sitemap.xml` for SEO?

**Comment by mikhin** [comment]
> Would you be up for a PR?

Yeah, I'm ready to get involved

> What's your motivation?

In my particular case, I need it for SEO, yes. I want to feed the file into Google Search Console:

<img width="1167" alt="Screenshot 2024-01-21 at 21 11 33" src="https://github.com/vikejs/vike/assets/38258318/7a79c82d-3d0d-483e-80da-f2093815ecf8">

**Comment by brillout** [comment]
👍 Looking forward to it.

**Comment by mikhin** [comment]
Could you suggest suitable internal APIs from Vike? What would it look like in theory?

I'm asking because I'm not familiar with the Vike structure.

**Comment by brillout** [comment]
See `runPrerender.ts`. If you prefer you can write a script you run after an one of the example's `$ vite build` (e.g. `/examples/{react,vue}-full` and I'll do the integration of your script into Vike. Whichever you prefer.

**Comment by Idrinth** [comment]
@mikhin if you want some code for a script that should work with minor adjustments feel free to take from https://github.com/Idrinth/react-file-based-routes/blob/master/src/generate-routes.ts - would be happy if vike would support sitemaps out of the box.

Required adjustments:
- point to dist instead of src
- look for .html files instead of .tsx
- possibly adjust the mapping object

**Comment by mikhin** [comment]
Sounds great, thanks for the tip!

**Comment by brillout** [comment]
Quoting @br0wsa https://github.com/vikejs/vike/issues/2234.

> I would be happy to [...] contribute to its development.

Yes, contribution welcome!

At least some rudimentary support for SEO purposes would be nice indeed. I guess simply generating `dist/client/sitemap.xml` with the list of URLs would be good enough for SEO purposes? IIRC / I guess search bots automatically try to fetch `/sitemap.xml` (no need to have a link to it).

> Allows configuration options for generation, such as update frequency and page priority.

Not sure I follow this one: aren't sitemaps always static? Not sure what you mean with page priority.

> Provides a simple user interface for downloading the XML sitemap file.

Interesting. What's the use case? 👀 

We can add built-in support for sitemap.xml into Vike core, or create a Vike extension, or both. If all features make sense for most users we can make it all built-in, but we can (also) create Vike extensions for tailored use cases.

**Comment by brillout** [comment]
https://github.com/Qalisa/vike-plugin-sitemap
